src/display.rs

```diff
@@ -252,51 +257,54 @@ fn get_output<'a>(
             Block::Permission => {
-                let s: &[ColoredString] = &[
+                block_vec.extend(vec![
                     meta.file_type.render(colors),
                     meta.permissions.render(colors),
-                ];
-                let res = ANSIStrings(s).to_string();
-                strings.push(ColoredString::from(res));
+                ]);
             }
-            Block::User => strings.push(meta.owner.render_user(colors)),
-            Block::Group => strings.push(meta.owner.render_group(colors)),
-            Block::Size => strings.push(meta.size.render(
-                colors,
-                &flags,
-                padding_rules[&Block::SizeValue],
-            )),
-            Block::SizeValue => strings.push(meta.size.render_value(colors, flags)),
-            Block::Date => strings.push(meta.date.render(colors, &flags)),
+            Block::User => block_vec.push(meta.owner.render_user(colors)),
+            Block::Group => block_vec.push(meta.owner.render_group(colors)),
+            Block::Size => {
+                let pad = if Layout::Tree == flags.layout && 0 == tree.0 && 0 == i {
+                    None
+                } else {
+                    Some(padding_rules[&Block::SizeValue])
+                };
+                block_vec.push(meta.size.render(colors, &flags, pad))
+            }
+            Block::SizeValue => block_vec.push(meta.size.render_value(colors, flags)),
+            Block::Date => block_vec.push(meta.date.render(colors, &flags)),
             Block::Name => {
-                let s: String =
-                    if flags.no_symlink.0 || flags.dereference.0 || flags.layout == Layout::Grid {
-                        ANSIStrings(&[
-                            meta.name.render(colors, icons, &display_option),
-                            meta.indicator.render(&flags),
-                        ])
-                        .to_string()
-                    } else {
-                        ANSIStrings(&[
-                            meta.name.render(colors, icons, &display_option),
-                            meta.indicator.render(&flags),
-                            meta.symlink.render(colors, &flags),
-                        ])
-                        .to_string()
-                    };
-
-                strings.push(ColoredString::from(s));
+                if flags.no_symlink.0 || flags.dereference.0 || flags.layout == Layout::Grid {
+                    block_vec.extend(vec![
+                        meta.name.render(colors, icons, &display_option),
+                        meta.indicator.render(&flags),
+                    ])
+                } else {
+                    block_vec.extend(vec![
+                        meta.name.render(colors, icons, &display_option),
+                        meta.indicator.render(&flags),
+                        meta.symlink.render(colors, &flags),
+                    ])
+                };
```

Comment:
We could probably simplify this.

```suggestion
                block_vec.extend(vec![
                    meta.name.render(colors, icons, &display_option),
                    meta.indicator.render(&flags),
                ]);
                if !(flags.no_symlink.0 || flags.dereference.0 || flags.layout == Layout::Grid) {
                    block_vec.push(meta.symlink.render(colors, &flags))
                }
```